Assets/Scripts/TorchStand.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TorchStand : MonoBehaviour
```

Comment:
The same story there.
Now when you removed interaction logic from player you should put it there.
Assign this script as interactable by adding inkeritance from interface
public class TorchStand : MonoBehaviour, IInteractable
This will force you to implement method form this so do it.
```cs
public void Interact(GameObject interacter)
{
  var playerHands  = interacter.GetComponent<PlayerHends>();
   // Copied form you script
   if (playerHands.IsHoldingTorch)
    {
        if (!HasTorch)
        {
            PutTorch(playerHands.IsTorchOn);
            playerHands.TakeOffTorch();
        }
    }
    else
    {
        playerHands.TakeTorch(torchStand.IsTorchOn);
        TakeTorch();
    }
}
```